core/trino-main/src/main/java/io/trino/server/security/oauth2/NimbusAirliftHttpClient.java

```diff
@@ -93,9 +93,9 @@ public <T> T execute(com.nimbusds.oauth2.sdk.Request nimbusRequest, Parser<T> pa
         request.addHeaders(headers.build());
 
         if (method.equals(POST) || method.equals(PUT)) {
-            String query = httpRequest.getQuery();
+            String query = httpRequest.getBody();
             if (query != null) {
-                request.setBodyGenerator(createStaticBodyGenerator(httpRequest.getQuery(), UTF_8));
+                request.setBodyGenerator(createStaticBodyGenerator(httpRequest.getBody(), UTF_8));
```

Comment:
nit: not related to this change but you could use `query` variable here.

Reply:
you are right.